tests/shared.rs

```diff
@@ -38,3 +38,31 @@ fn two_threads() {
 fn many_threads() {
     send_shared_oneshot_and_wait_on_multiple_threads(1000);
 }
+
+#[test]
+fn drop_on_one_task_ok() {
```

Comment:
This test case passes for me even *before* applying the rest of this commit, so I doubt that it is providing the coverage that we want.